src/Workspace/index.jsx

```diff
@@ -733,20 +736,12 @@ class Workspace extends React.Component {
           }
         </div>
       );
-    } if (this.state.defaultWorkspace && this.state.connectedStatus) {
-      // If this commons does not use Hatchery to spawn workspaces, then this
-      // default workspace is shown.
-      return (
-        <div className='workspace__default'>
-          <iframe
-            title='Workspace'
-            frameBorder='0'
-            className='workspace__iframe'
-            src={workspaceUrl}
-            onLoad={this.oniframeLoad}
-          />
-        </div>
-      );
+    } if (this.state.connectedStatus && !this.state.hasWorkspaceAccess) {
+      if (isEnabled('workspaceRegistration')) {
+        console.log('This is navigated to workspace register page');
+        return <Redirect to='/workspace/register' />;
+      }
+      return <Redirect to={workspaceErrorUrl} />;
```

Comment:
why not redirect to `Error403Url` in this case?

Reply:
Actually, if we can tune the `hasWorkspaceAccess` value a bit finer, maybe we can do something like this:

- If https://github.com/uc-cdis/data-portal/pull/1067/files#diff-f907a00fd9e7338bc39b77dc677d401f5dad62c284b04a4f1d94345854432f24R130 gives a 403 error AND workspace registration is enabled: redirect to ws reg page
- If https://github.com/uc-cdis/data-portal/pull/1067/files#diff-f907a00fd9e7338bc39b77dc677d401f5dad62c284b04a4f1d94345854432f24R130 gives a 403 error AND workspace registration is NOT enabled: redirect to error 403 page
- If https://github.com/uc-cdis/data-portal/pull/1067/files#diff-f907a00fd9e7338bc39b77dc677d401f5dad62c284b04a4f1d94345854432f24R130 gives a NON-403 error: redirect to the generic workspace error page

WDYT about this approach? 🤔

Reply:
That seems like a good way to go! Just to make sure I understand it right, 
we redirect the users to- 
1.  `/no-workspace-access` page for any `non-403` error,
2. `/workspace/register` page for a `403 error` with `workspaceRegistration` flag enabled, 
3. but if the response's status is `403` we then just update the `this.setState({ hasWorkspaceAccess: false });` and redirect users to the `/403Error` page [here](https://github.com/uc-cdis/data-portal/blob/e543a1ec4db83b3f1348a3a490e92b959c375c01/src/Workspace/index.jsx#L747-L747)

Reply:
yeah, although we should rename the path the generic workspace error page to something else, because in that case, it is not because of no workspace access

maybe the 403 error page should take the `/no-workspace-access` path and the generic workspace error page should be `/workspace-error`

Reply:
Update: nginx redirects the traffic to workspace error page now, so we won't see 403 from portal side

We can let nginx stop from doing redirection, but it is out of the scope of this ticket

Reply:
Well, I still don't think the 403 page should be `/no-workspace-access`, for instance, I guess, thor has nothing to do with workspaces but I still see it being navigated to `/no-workspace-access` [here](https://github.com/uc-cdis/cloud-automation/blob/master/kube/services/revproxy/gen3.nginx.conf/thor-service.conf#L18-L18)

Reply:
yeah that is correct, we don't have a generic 403 error page at that time and everything was being redirected to ws error page, not good